examples/cluster-with-resources/main.tf

```diff
@@ -0,0 +1,39 @@
+variable "server_side_planning" {
+  type = bool
+}
+
+provider "azurerm" {
+  version = ">=2.20.0"
+  features {}
+}
+
+provider "kubernetes-alpha" {
+  server_side_planning = false #var.server_side_planning
+
+  host = module.cluster.host
+  cluster_ca_certificate = module.cluster.cluster_ca_certificate
+  client_certificate = module.cluster.client_certificate
+  client_key = module.cluster.client_key
+}
+
+resource "random_id" "uniq" {
+  keepers = {
+    api_host = module.cluster.host
+  }
+
+  byte_length = 2
+}
+
+module "cluster" {
+  source = "./cluster"
+}
+
+module "manifests" {
+  source = "./manifests"
+
+  depends_on = [module.cluster]
```

Comment:
This `depends_on` requires terraform version 0.13 and above. It would probably be good to mention this on the example's documentation because otherwise you get this error:

```
$ terraform validate

Error: Reserved argument name in module block

  on main.tf line 34, in module "manifests":
  34:   depends_on = [module.cluster]

The name "depends_on" is reserved for use in a future version of Terraform.
```

Reply:
hmm, but also, I can't seem to initialize this workspace with terraform 0.13.

```
$ terraform0.13.0-beta3 init
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/local...
- Finding latest version of hashicorp/random...
- Finding hashicorp/azurerm versions matching ">= 2.20.0"...
- Finding latest version of hashicorp/kubernetes-alpha...
- Installing hashicorp/local v1.4.0...
- Installed hashicorp/local v1.4.0 (signed by HashiCorp)
- Installing hashicorp/random v2.3.0...
- Installed hashicorp/random v2.3.0 (signed by HashiCorp)
- Installing hashicorp/azurerm v2.21.0...
- Installed hashicorp/azurerm v2.21.0 (signed by HashiCorp)
source
Error: Failed to install provider

Error while installing hashicorp/kubernetes-alpha: provider registry
registry.terraform.io does not have a provider named
registry.terraform.io/hashicorp/kubernetes-alpha
```

This is my working directory:
```
$ pwd
/home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes-alpha/examples/cluster-with-resources
```

Is there a specific version I should use to test this?

Reply:
> This `depends_on` requires terraform version 0.13 and above. It would probably be good to mention this on the example's documentation because otherwise you get this error:
> 
> ```
> $ terraform validate
> 
> Error: Reserved argument name in module block
> 
>   on main.tf line 34, in module "manifests":
>   34:   depends_on = [module.cluster]
> 
> The name "depends_on" is reserved for use in a future version of Terraform.
> ```

Actually I'll get rid of that. It's not even necessary since the dependency can be established through values passed between the two modules.

Reply:
You can test with both 0.12.x and 0.13.x.

If you go for 0.13, then you need to install the local provider binary in a different location than where it's required for 0.12.x. This is one of the major changes in 0.13 and so far it's only documented clearly over here: https://github.com/hashicorp/terraform/tree/guide-v0.13-beta/provider-sources#provider-plugins-in-the-local-filesystem

On my machine (OSX) the binary needs to be located here: $HOME/.terraform.d/plugins/registry.terraform.io/hashicorp/kubernetes-alpha/0.0.1/darwin_amd64/terraform-provider-kubernetes-alpha_0.0.1_darwin_amd64

> hmm, but also, I can't seem to initialize this workspace with terraform 0.13.
> 
> ```
> $ terraform0.13.0-beta3 init
> Initializing modules...
> 
> Initializing the backend...
> 
> Initializing provider plugins...
> - Finding latest version of hashicorp/local...
> - Finding latest version of hashicorp/random...
> - Finding hashicorp/azurerm versions matching ">= 2.20.0"...
> - Finding latest version of hashicorp/kubernetes-alpha...
> - Installing hashicorp/local v1.4.0...
> - Installed hashicorp/local v1.4.0 (signed by HashiCorp)
> - Installing hashicorp/random v2.3.0...
> - Installed hashicorp/random v2.3.0 (signed by HashiCorp)
> - Installing hashicorp/azurerm v2.21.0...
> - Installed hashicorp/azurerm v2.21.0 (signed by HashiCorp)
> source
> Error: Failed to install provider
> 
> Error while installing hashicorp/kubernetes-alpha: provider registry
> registry.terraform.io does not have a provider named
> registry.terraform.io/hashicorp/kubernetes-alpha
> ```
> 
> This is my working directory:
> 
> ```
> $ pwd
> /home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes-alpha/examples/cluster-with-resources
> ```
> 
> Is there a specific version I should use to test this?

Reply:
Thanks, that worked for me! Here's the linux equivalent in case someone else runs into this.

```
mkdir -p ~/.terraform.d/plugins/registry.terraform.io/hashicorp/kubernetes-alpha/0.1.0/linux_amd64

go build -o ~/.terraform.d/plugins/registry.terraform.io/hashicorp/kubernetes-alpha/0.1.0/linux_amd64/terraform-provider-kubernetes-alpha_0.1.0_linux_amd64
```